### PR TITLE
Fix pkg-config invocation during cross-compilation

### PR DESCRIPTION
**Pull request by aol-nnov**
Also, honour system locations for `--with-sdl` and `--with-ffmpeg`

Closes https://github.com/pjsip/pjproject/issues/4152

### CODE REVIEW

File: aconfigure.ac [hunk 1]
```diff
@@ -2021,7 +2023,7 @@ AC_ARG_ENABLE(ssl,
     		    fi
 
         	    AC_CHECK_PROGS(PKG_CONFIG,
-                       $host-pkg-config pkg-config "python pkgconfig.py",
+                       $host_cpu-$host_os-pkg-config pkg-config "python pkgconfig.py",
```
Comment (sauwming) [inline aconfigure.ac:2026]: Googling around (such as:
https://autotools.info/pkgconfig/cross-compiling.html)
, `$host-pkg-config` seems to be commonly used as well, so better not remove it.

**Review by sauwming — COMMENTED** [review]
Like the other PR, also need to include the `aconfigure`.

File: aconfigure.ac (hunk 1 above)
Comment (aol-nnov) [reply aconfigure.ac:2026]: Good point, will keep that, too.

**Comment by aol-nnov** [comment]
> Like the other PR, also need to include the aconfigure.

Agreed, let's finish that one and then I will rebase this on top and regenerate `aconfigure` as well.

File: aconfigure.ac (hunk 1 above)
Comment (aol-nnov) [reply aconfigure.ac]: done

**Comment by aol-nnov** [comment]
@sauwming I've discovered two `AC_CHECK_PROGS(PKG_CONFIG,...)` calls along the `aconfigure.ac`, so I decided to combine them and move closer to the top of the file.

Also, as I mentioned earlier, system locations for `--with-sdl` and `--with-ffmpeg` are now honoured if those options are provided without arguments.

`aconfigure` is also re-generated according to the introduced changes. Please review.

**Review by sauwming — APPROVED** [review]

**Review by nanangizz — APPROVED** [review]

**Comment by aol-nnov** [comment]
Thank you for the prompt responses @sauwming @nanangizz ! It's a pleasure to work with you!

**Pull request merged**